torch/_meta_registrations.py

```diff
@@ -2695,191 +2696,527 @@ def meta_addbmm(self, batch1, batch2, *, beta=1, alpha=1):
+            for i in range(len(input))
+        ]
+
+    @register_meta(inplace)
+    def _inplace_meta(
+        input: List[Tensor],
+        tensors1: List[Tensor],
+        tensors2: List[Tensor],
+        scalars: Tensor,
+    ):
+        _common(input, tensors1, tensors2)
+        return None
+
+    return _func_meta, _inplace_meta
+
+
+def _foreach_ternary_op(func, inplace, op):
+    def _common(input: List[Tensor], tensors1: List[Tensor], tensors2: List[Tensor]):
+        _check_foreach_api_restrictions_pointwise(input, tensors1, tensors2)
+
+    @register_meta(func)
+    def _func_meta(input: List[Tensor], tensors1: List[Tensor], tensors2: List[Tensor]):
+        _common(input, tensors1, tensors2)
+        return [op(input[i], tensors1[i], tensors2[i]) for i in range(len(input))]
+
+    @register_meta(inplace)
+    def _inplace_meta(
+        input: List[Tensor], tensors1: List[Tensor], tensors2: List[Tensor]
+    ):
+        _common(input, tensors1, tensors2)
+        return None
+
+    return _func_meta, _inplace_meta
+
+
+_foreach_binary_op_list_alpha(
+    aten._foreach_add.List, aten._foreach_add_.List, torch.add
+)
+_foreach_binary_op_list_alpha(
+    aten._foreach_sub.List, aten._foreach_sub_.List, torch.sub
+)
+# XXX: lerp: weight is not a kwarg
+_foreach_binary_op_list_alpha(
+    aten._foreach_lerp.Scalar, aten._foreach_lerp_.Scalar, torch.lerp, is_kwarg=False
+)
+
+
+_foreach_binary_op_tensor(
+    aten._foreach_mul.Tensor, aten._foreach_mul_.Tensor, torch.mul
+)
+
+
+_foreach_binary_op_scalar(
+    aten._foreach_add.Scalar, aten._foreach_add_.Scalar, torch.add
+)
+_foreach_binary_op_scalar(
+    aten._foreach_sub.Scalar, aten._foreach_sub_.Scalar, torch.sub
+)
+_foreach_binary_op_scalar(
+    aten._foreach_mul.Scalar, aten._foreach_mul_.Scalar, torch.mul
+)
+_foreach_binary_op_scalar(
+    aten._foreach_div.Scalar, aten._foreach_div_.Scalar, torch.div
+)
+_foreach_binary_op_scalar(
+    aten._foreach_maximum.Scalar, aten._foreach_maximum_.Scalar, torch.clamp_min
+)
+_foreach_binary_op_scalar(
+    aten._foreach_minimum.Scalar, aten._foreach_minimum_.Scalar, torch.clamp_max
+)
+_foreach_binary_op_scalar(
+    aten._foreach_clamp_min.Scalar, aten._foreach_clamp_min_.Scalar, torch.clamp_min
+)
+_foreach_binary_op_scalar(
+    aten._foreach_clamp_max.Scalar, aten._foreach_clamp_max_.Scalar, torch.clamp_max
+)
+_foreach_binary_op_scalar(
+    aten._foreach_pow.Scalar, aten._foreach_pow_.Scalar, torch.pow
+)
+# _foreach_binary_op_scalar(aten._foreach_norm.Scalar, None, torch.linalg.vector_norm, has_inplace=False)
+
+
+# XXX: same as _foreach_binary_op_scalar, but the second argument is named ord and has no inplace overload
+# @register_meta(aten._foreach_norm.Scalar)
+# def _func_meta(tensors: List[Tensor], ord: NumberType = 2):
+#     _check_foreach_api_restrictions_unary(tensors)
+#     return [torch.linalg.vector_norm(tensors[i], ord) for i in range(len(tensors))]
```

Comment:
Need to figure out how to define this consistently. Plus figure out the issue on CUDA.